### PR TITLE
Fix `report()` error message responsibility for a missing node or line number

### LINKED ISSUE #7172: Fix unexpected error message in `report()`
### What minimal example or steps are needed to reproduce the bug?

Empty sass file

### What minimal configuration is needed to reproduce the bug?

```json
{
    "customSyntax": "postcss-sass",
    "extends": "stylelint-config-standard"
}
```

### How did you run Stylelint?

```shell
npx stylelint -f verbose "**/*.sass"
```

### Which Stylelint-related dependencies are you using?

```json
  "devDependencies": {
    "postcss-sass": "^0.5.0",
    "stylelint": "^15.10.3",
    "stylelint-config-standard": "^34.0.0"
  }
```

### What did you expect to happen?

No problem to report

### What actually happened?

Stylelint crashed:
```
Error: You must pass either a node or a line number
    at report (/home/georges/data/tmp/dev/stylelint-empty-file/node_modules/stylelint/lib/utils/report.js:42:9)
    at /home/georges/data/tmp/dev/stylelint-empty-file/node_modules/stylelint/lib/rules/no-empty-source/index.js:32:3
    at /home/georges/data/tmp/dev/stylelint-empty-file/node_modules/stylelint/lib/lintPostcssResult.js:121:8
    at Array.map (<anonymous>)
    at lintPostcssResult (/home/georges/data/tmp/dev/stylelint-empty-file/node_modules/stylelint/lib/lintPostcssResult.js:111:18)
    at lintSource (/home/georges/data/tmp/dev/stylelint-empty-file/node_modules/stylelint/lib/lintSource.js:110:8)
    at async /home/georges/data/tmp/dev/stylelint-empty-file/node_modules/stylelint/lib/standalone.js:211:27
    at async Promise.all (index 1)
    at async standalone (/home/georges/data/tmp/dev/stylelint-empty-file/node_modules/stylelint/lib/standalone.js:254:22)
```

### Do you have a proposal to fix the bug?

Stylelint fails to process an empty sass file.

Again, I am not sure if it comes from stylelint or postcss-sass plugin. It works properly with the `postcss-scss` plugin though.

Also, the error message is not very practical, it does not even specify which file made stylelint crash, even with the `-f verbose` option. Any way to improve debugging messages?

### PR DESCRIPTION
**Pull request by ybiquitous**
> Which issue, if any, is this issue related to?

Closes #7172

> Is there anything in the PR that needs further explanation?

Including a rule name in the error message helps us understand why an error happens.

### CODE REVIEW

File: lib/utils/report.mjs [hunk 1]
```diff
@@ -43,7 +43,7 @@ export default function report(problem) {
 	const startLine = line || (start && start.line);
 
 	if (!startLine) {
-		throw new Error('You must pass either a node or a line number');
+		throw new Error(`You must pass either a node or a line number (rule: ${ruleName})`);
```
Comment (jeddy3) [inline lib/utils/report.mjs:46]: Am I right in thinking that only the (custom) rule author can resolve this issue and not the user? If so, let's rewrite the error to address the original problem: lack of clarity around the message.

```suggestion
		throw new Error(`The "${ruleName}" (custom) rule failed to pass either a node or a line number to the `report()` function.`);
```

File: .changeset/weak-maps-tease.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"stylelint": patch
+---
+
+Fixed: unexpected error message in `report()`
```
Comment (jeddy3) [inline .changeset/weak-maps-tease.md:5]:
```suggestion
Fixed: `report()` error message clarity for missing node or line number
```

File: lib/utils/report.mjs (hunk 1 above)
Comment (ybiquitous) [reply lib/utils/report.mjs:46]: Nice catch! Your suggestion is much better. 👍🏼

**Review by jeddy3 — APPROVED** [review]
LGTM, thank you.

**Pull request merged**